### PR TITLE
Add OpenTelemetry traces support

### PR DESCRIPTION
**Pull request by topikachu**
Add OpenTelemetry traces support.

1. When [Jenkins OpenTelemetry](https://plugins.jenkins.io/opentelemetry/) is available, retrieve span from opentelemetry plugin.
2. Send [W3C traceparent header](https://www.w3.org/TR/trace-context/#traceparent-header) when span is available.
3. Modernize the plugin baseline to integrate with opentelemetry plugin
### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

**Comment by topikachu** [comment]
@cashlalala Could you please take a look at my change?

**Comment by cashlalala** [comment]
@topikachu Thanks for your contribution! I'll take a look these two weeks, kind of busy recently.

File: pom.xml [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
-		<version>3.50</version>
+		<version>4.50</version>
+		<relativePath />
 	</parent>
 
   <properties>
-    <jenkins.version>1.653</jenkins.version>
-    <java.level>8</java.level>
+    <jenkins.version>2.346.3</jenkins.version>
+    <java.level>11</java.level>
```
Comment (cashlalala) [inline pom.xml]: Is that possible to make the baseline version upgrade decoupled with open-telemetry related changes? 
This is a good feature and legacy Jenkins version should be able to use it as well.

**Review by cashlalala — CHANGES_REQUESTED** [review]
Another PR for the baseline version upgrade would be appreciated

File: pom.xml (hunk 1 above)
Comment (topikachu) [reply pom.xml]: @cashlalala 
I meet a lot of dependency errors when I use the original baseline version.
I believe this is introduced by the opentelemetry plugin. In other words, to use it, users must upgrade their Jenkins to meet the requirements.
I use the same baseline version as the opentelemetry ones.

Any idea?

Comment (topikachu) [reply pom.xml]: Please see #85

Comment (cashlalala) [reply pom.xml]: I was thinking that you might find the preivous version of open telemetry plugin with compatibility of jenkins 1.653 
then upgrade the baseline for newer master and open telemetry plugin.

Comment (topikachu) [reply pom.xml]: @cashlalala 
Please check https://github.com/jenkinsci/opentelemetry-plugin/blob/105ad4bf8da347dbc26bb42e55c97404fdef2884/pom.xml#L22C1-L23C1
It seems opentelemetry chose [2.235.1](https://www.jenkins.io/changelog-stable/#v2.235.1) from the beginning.

Comment (topikachu) [reply pom.xml]: I may redo the change. 
Perhaps I can get rid of the opentelemetry plugin completely!

**Pull request merged**